sw/device/lib/otbn.h

```diff
@@ -0,0 +1,248 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_otbn.h"
+
+/**
+ * @file
+ * @brief OpenTitan Big Number Accelerator (OTBN) driver
+ */
+
+/**
+ * Information about an embedded OTBN application image.
+ *
+ * All pointers reference data in the normal CPU address space.
+ *
+ * Use `OTBN_USE_APP()` together with `OTBN_APP_T_INIT()` to initialize this
+ * structure.
+ */
+typedef struct otbn_app {
+  /**
+   * Start of OTBN instruction memory.
+   */
+  const char *imem_start;
+  /**
+   * End of OTBN instruction memory.
+   */
+  const char *imem_end;
+  /**
+   * Start of OTBN data memory.
+   */
+  const char *dmem_start;
+  /**
+   * End of OTBN data memory.
+   */
+  const char *dmem_end;
+} otbn_app_t;
+
+/**
+ * A pointer to a symbol in OTBN's instruction or data memory.
+ *
+ * Use `OTBN_USE_PTR()` together with `OTBN_PTR_T_INIT()` to initialize this
+ * structure.
+ */
+typedef struct otbn_ptr {
+  /**
+   * Address of the symbol in normal CPU address space.
+   */
+  const char *ptr_in_cpu_addr_space;
+} otbn_ptr_t;
```

Comment:
I think this is the type of thing we'd normally just make a typedef in a DIF. I think newtypes are great but maybe it's worth being consistent?

Reply:
Removed the struct.